snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/SnomedRelationshipUpdateRequest.java

```diff
@@ -15,80 +15,98 @@
  */
 package com.b2international.snowowl.snomed.datastore.request;
 
-import java.util.Objects;
 import java.util.Set;
 
 import javax.validation.constraints.Max;
 import javax.validation.constraints.Min;
 
+import com.b2international.commons.exceptions.BadRequestException;
 import com.b2international.snowowl.core.date.EffectiveTimes;
 import com.b2international.snowowl.core.domain.TransactionContext;
 import com.b2international.snowowl.snomed.common.SnomedRf2Headers;
+import com.b2international.snowowl.snomed.core.domain.RelationshipValue;
 import com.b2international.snowowl.snomed.datastore.index.entry.SnomedConceptDocument;
 import com.b2international.snowowl.snomed.datastore.index.entry.SnomedRelationshipIndexEntry;
+import com.google.common.base.Strings;
 import com.google.common.collect.ImmutableSet;
-import com.google.common.collect.ImmutableSet.Builder;
 
 /**
  * @since 4.5
  */
 public final class SnomedRelationshipUpdateRequest extends SnomedComponentUpdateRequest {
 
-	@Min(0)
-	@Max(Integer.MAX_VALUE)
+	// Not @NotEmpty, update of type ID is optional
+	private String typeId;
+
+	// Not @NotEmpty, update of destination ID is optional
+	private String destinationId;
+
+	// Not @NotNull, update of relationship value is optional
+	private RelationshipValue value;
+
+	@Min(0)	@Max(Integer.MAX_VALUE)
```

Comment:
A bit of inconsistency between the create and update requests. Create allows only `Byte.MAX_VALUE` while update allows `Integer.MAX_VALUE` `groups` and `unionGroups`.